daml/Marketplace/Trading.daml

```diff
@@ -153,44 +109,93 @@ template Service
     maintainer key._1
 
     controller customer can
-      nonconsuming RequestCreateOrder : (ContractId CreateOrderRequest, ContractId AssetDeposit)
+      nonconsuming RequestCreateOrder : (ContractId Order, ContractId CreateOrderRequest, ContractId AssetDeposit)
```

Comment:
So I had split this into `Order` and `Trade` files, but @georg-da nuked them. I think that would make it tidier so we don't have to write `OrderDetails`, `OrderRequest`, `OrderStatus` etc. But maybe he was just rushed. Let's wait til he gets back to see what he thinks of that.

Reply:
This is likely due to cyclic dependency hell - I tried to split them up but found its that time consuming that I gave up and left it in its current form. I'd be happy to tackle it again later once the trading features are at a more mature stage.